docs/ERC721.md

```diff
@@ -0,0 +1,686 @@
+    func name() -> (name: felt):
+    end
+
+    func symbol() -> (symbol: felt):
+    end
+
+    func tokenURI(token_id: Uint256) -> (uri_len: felt, uri: felt*):
+    end
+
+--------------- IERC721 ---------------
+
+    func balanceOf(owner: felt) -> (balance: Uint256):
+    end
+
+    func ownerOf(token_id: Uint256) -> (owner: felt):
+    end
+
+    func safeTransferFrom(
+        _from: felt, 
+        to: felt, 
+        token_id: Uint256, 
+        data_len: felt,
+        data: felt*
+    ):
+
+    func transferFrom(_from: felt, to: felt, token_id: Uint256):
+    end
+
+    func approve(approved: felt, token_id: Uint256):
+    end
+
+    func setApprovalForAll(operator: felt, approved: felt):
+    end
+
+    func getApproved(token_id: Uint256) -> (approved: felt):
+    end
+
+    func isApprovedForAll(owner: felt, operator: felt) -> (is_approved: felt):
+    end
+
+--------------- IERC165 ---------------
+
+    func supportsInterface(interface_id: felt) -> (success: felt):
+    end
+
+```
+
+### IERC721_Enumerable
+
+```jsx
+@contract_interface
+namespace IERC721_Enumerable:
+    func totalSupply() -> (totalSupply: Uint256):
+    end
+
+    func tokenByIndex(index: Uint256) -> (token_id: Uint256):
+    end
+
+    func tokenOfOwnerByIndex(owner: felt, index: Uint256) -> (token_id: Uint256):
+    end
+
+--------------- IERC721 ---------------
+
+    func balanceOf(owner: felt) -> (balance: Uint256):
+    end
+
+    func ownerOf(token_id: Uint256) -> (owner: felt):
+    end
+
+    func safeTransferFrom(
+        _from: felt, 
+        to: felt, 
+        token_id: Uint256, 
+        data_len: felt,
+        data: felt*
+    ):
+
+    func transferFrom(_from: felt, to: felt, token_id: Uint256):
+    end
+
+    func approve(approved: felt, token_id: Uint256):
+    end
+
+    func setApprovalForAll(operator: felt, approved: felt):
+    end
+
+    func getApproved(token_id: Uint256) -> (approved: felt):
+    end
+
+    func isApprovedForAll(owner: felt, operator: felt) -> (is_approved: felt):
+    end
+
+--------------- IERC165 ---------------
+
+    func supportsInterface(interface_id: felt) -> (success: felt):
+    end
+
+```
+
+### IERC721_Receiver
+
+```jsx
+@contract_interface
+namespace IERC721_Receiver:
+    func onERC721Received(
+        operator: felt,
+        _from: felt,
+        token_id: Uint256,
+        data_len: felt
+        data: felt*
+    ) -> (selector: felt):
+    end
+```
+
+### IERC165
+```jsx
+@contract_interface
+namespace IERC165:
+    func supportsInterface(interface_id: felt) -> (success: felt):
+    end
+```
+
+### ERC721 Compatibility
+
+Although StarkNet is not EVM compatible, this implementation aims to be as close as possible to the ERC721 standard in the following ways:
+
+- it uses Cairo's `uint256` instead of `felt`
+- it returns `1` as true/success to imitate a `bool`
+- it makes use of Cairo's short strings to simulate `name` and `symbol`
+- it will emit events once they're implemented on StarkNet
```

Comment:
Some of this we might want to abstract away from here and ERC20 docs, then link to it since it will probably change over time.